packages/grid/_modules_/grid/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -0,0 +1,252 @@
+import * as React from 'react';
+import ownerDocument from '@material-ui/core/utils/ownerDocument';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridApiMethod } from '../../root/useGridApiMethod';
+import { GridPrintExportApi } from '../../../models/api/gridPrintExportApi';
+import { useLogger } from '../../utils/useLogger';
+import { useGridSelector } from '../core/useGridSelector';
+import { visibleSortedGridRowsSelector } from '../filter/gridFilterSelector';
+import { useGridState } from '../core/useGridState';
+import { GridComponentProps } from '../../../GridComponentProps';
+import { GridPrintExportOptions } from '../../../models/gridExport';
+import { allGridColumnsSelector } from '../columns/gridColumnsSelector';
+
+type PrintWindowOnLoad = (
+  printWindow: HTMLIFrameElement,
+  options?: Pick<
+    GridPrintExportOptions,
+    'copyStyles' | 'bodyClass' | 'pageStyle' | 'hideToolbar' | 'hideFooter'
+  >,
+) => void;
+
+export const useGridPrintExport = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'rowHeight'>,
+): void => {
+  const logger = useLogger('useGridPrintExport');
+  const [, setGridState, forceUpdate] = useGridState(apiRef);
+  const visibleSortedRows = useGridSelector(apiRef, visibleSortedGridRowsSelector);
+  const columns = useGridSelector(apiRef, allGridColumnsSelector);
+  const doc = React.useRef<Document | null>(null);
+  const previousGridState = React.useRef<any>(null);
+  const previousHiddenColumns = React.useRef<string[]>([]);
+
+  React.useEffect(() => {
+    doc.current = ownerDocument(apiRef!.current.rootElementRef!.current as HTMLElement);
```

Comment:
The non-null operator is not necessary here

```suggestion
    doc.current = ownerDocument(apiRef.current.rootElementRef!.current!);
```